src/components/Modals/BuySell/components/AssetSearch/AssetSearch.tsx

```diff
@@ -0,0 +1,143 @@
+import { SearchIcon } from '@chakra-ui/icons'
+import { Box, Center, Input, InputGroup, InputLeftElement } from '@chakra-ui/react'
+import { Asset } from '@shapeshiftoss/types'
+import axios from 'axios'
+import { getConfig } from 'config'
+import { concat, flatten, uniqBy } from 'lodash'
+import { FormEvent, useEffect, useMemo, useState } from 'react'
+import { useForm } from 'react-hook-form'
+import { CircularProgress } from 'components/CircularProgress/CircularProgress'
+
+import { BuySellAction } from '../../BuySell'
+import { AssetList } from './AssetList'
+import { filterAssetsBySearchTerm } from './helpers/filterAssetsBySearchTerm'
+
+type AssetSearchProps = {
+  onClick: (asset: any) => void
+  type: BuySellAction
+}
+
+export const AssetSearch = ({ onClick, type = BuySellAction.Buy }: AssetSearchProps) => {
+  const [filteredAssets, setFilteredAssets] = useState<Asset[]>([])
+  const { register, watch } = useForm<{ search: string }>({
+    mode: 'onChange',
+    defaultValues: {
+      search: ''
+    }
+  })
+
+  const searchString = watch('search')
+  const searching = useMemo(() => searchString.length > 0, [searchString])
+
+  const [loading, setLoading] = useState(false)
+  const [currentAssets, setCurrentAssets] = useState<any[]>([])
+
+  const getCoinifySupportedCurrencies = async () => {
+    try {
+      const { data } = await axios.get(getConfig().REACT_APP_GEM_COINIFY_SUPPORTED_COINS)
+      return data
+    } catch (e: any) {
+      console.error(e)
+    }
+  }
+
+  const getWyreSupportedCurrencies = async () => {
+    try {
+      const { data } = await axios.get(getConfig().REACT_APP_GEM_WYRE_SUPPORTED_COINS)
+      return data
+    } catch (e: any) {
+      console.error(e)
+    }
+  }
+  //Filter for the assets you can buy
+  const buyFilter = (asset: { transaction_direction: string }) =>
+    asset.transaction_direction === 'bank_blockchain' ||
+    asset.transaction_direction === 'card_blockchain'
+
+  // Filter for the assets you can sell
+  const sellFilter = (asset: { transaction_direction: string }) =>
+    asset.transaction_direction === 'blockchain_bank'
+
+  //Filter and merge function
+  const filterAndMerge = useMemo(
+    () => (coinifyList: any[], wyreList: any[], key: string | number, filter: any) => {
+      const list1 = coinifyList
+        .filter(filter)
+        .map((list: { [x: string]: { currencies: any } }) => list[key].currencies)
```

Comment:
If filter functions are typed properly, `list` becomes typed implicitly so there will be no need for typing.

And again, if having abstracted types (`type x = {}`) and reusing them, you shouldn't have to type this with an incomplete subset of the full type of the response from `REACT_APP_GEM_COINIFY_SUPPORTED_COINS`
[Typescript Playground - Demo](https://www.typescriptlang.org/play?#code/C4TwDgpgBAKhDOwoF4oG8BQVtQGYHt8AuKAcgCMBDAJ1IwF8MMBjfAO0SmAWBLkQDaAXRRQBmHHkIkKNOvSFNWHJLgCWAG27UAYm1EAKAB58eAShQA+KACFCGiJTbGzS9p26IYAd3yjPwAB06loQ1AYh2npmgQC2lGDGVkZmQA)